client/src/components/topics/topic-tabs.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
+import Dialog from "@material-ui/core/Dialog";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import MenuItem from "@material-ui/core/MenuItem";
+import FormControl from "@material-ui/core/FormControl";
+import { ArrowDropDown, History } from "@material-ui/icons";
+import { State, TopicQuestions } from "types";
+
+import "styles/layout.css";
+import { useSelector } from "react-redux";
+import { Button, DialogActions, Paper, Tab, Tabs } from "@material-ui/core";
+import { shouldDisplayPortrait } from "pages";
+import { isMobile } from "react-device-detect";
+import { ChangeEvent } from "react";
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    container: {
+      display: "flex",
+      flexWrap: "wrap",
+    },
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 120,
+    },
+    root: {
+      flexGrow: 1,
+    },
+  })
+);
+
+function TopicTabs(props: {
+  topicQuestions: TopicQuestions[];
+  onTopicSelected: (topic: string) => void;
+  showHistoryTab: boolean;
+}): JSX.Element {
+  const { topicQuestions, onTopicSelected, showHistoryTab } = props;
+
+  const classes = useStyles();
+  const [open, setOpen] = useState<boolean>(false);
+  const [value, setValue] = React.useState<number>(0);
+  const curTopic = useSelector<State, string>((s) => s.curTopic);
+  const curMentor = useSelector<State, string>((state) => state.curMentor);
+
+  const firstTopic = useSelector<State, string>((state) => {
+    return state.mentorsById[curMentor]?.mentor?.topics[0]?.name || "";
+  });
+
+  const onChange = (
+    e: ChangeEvent<Record<string, unknown>>,
+    newValue: number
+  ) => {
+    setValue(newValue);
+  };
+
+  const onClickOpen = () => {
+    setOpen(true);
+  };
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const onTopicClick = (topic: string) => {
+    onTopicSelected(topic);
+    handleClose();
+  };
+
+  const topics = topicQuestions.map(({ topic }, index) =>
+    topic !== "History" ? (
+      <MenuItem
+        key={index}
+        value={index}
+        onClick={() => onTopicClick(topic)}
+        data-test={curTopic === topic ? topic : null}
+        data-cy="topic-opt-item"
+      >
+        {topic}
+      </MenuItem>
+    ) : null
+  );
+
+  const mobileView = (
+    <div style={{ width: "100%" }}>
+      <Paper className={classes.root} style={{ width: "100%" }}>
+        <Tabs
+          value={value}
+          TabIndicatorProps={{
+            style: { background: "#ddd", height: "10px" },
```

Comment:
all the places where we define colors inline in `style` elements...is there a reason not to define a class and then do this via css?

I think there *IS* a strong argument to put the `height` inline here, but the colors seems like would be better to externalize in a css file to make it easier to restyle the app?

Reply:
Most of the inline-css style moved to css files: https://github.com/mentorpal/mentor-client/pull/87/commits/9e17df685b96c06aa74ff3c59dcca9316c110ada

Reply:
TabIndicatorProps has to be inline because it's a prop from material-UI. This is the bottom-line color for the selected tab